src/content/containers/TimelineView.js

```diff
@@ -5,46 +5,55 @@
 // @flow
 import React, { PureComponent } from 'react';
 import { connect } from 'react-redux';
-import { getThreads, getThreadOrder } from '../reducers/profile-view';
-import actions from '../actions';
+import { getThreads } from '../reducers/profile-view';
+import { getThreadOrder, getHiddenThreads } from '../reducers/url-state';
+import { changeThreadOrder } from '../actions/profile-view';
 import FlameChartSettings from '../components/FlameChartSettings';
 import TimelineFlameChart from './TimelineFlameChart';
 import TimelineMarkers from './TimelineMarkers';
 import Reorderable from '../components/Reorderable';
 import { withSize } from '../with-size';
 
-import type { Thread } from '../../common/types/profile';
+import type { State } from '../reducers/types';
+import type { Thread, ThreadIndex } from '../../common/types/profile';
 
 require('./TimelineView.css');
 
 type Props = {
   threads: Thread[],
-  threadOrder: number[],
+  threadOrder: ThreadIndex[],
+  hiddenThreads: ThreadIndex[],
   height: number,
-  changeThreadOrder: any => any,
+  changeThreadOrder: typeof changeThreadOrder,
 };
 
 class TimlineViewTimelinesImpl extends PureComponent {
 
   props: Props
 
-  getScrollElement: () => ?HTMLElement
-  scrollElement: ?HTMLElement
+  _scrollElement: ?HTMLElement
 
   constructor(props: Props) {
     super(props);
-    this.getScrollElement = () => this.scrollElement;
+    (this: any)._getScrollElement = this._getScrollElement.bind(this);
+    (this: any)._setScrollElementRef = this._setScrollElementRef.bind(this);
+  }
+
+  _getScrollElement(): ?HTMLElement {
+    return this._scrollElement;
+  }
+
+  _setScrollElementRef(element: HTMLElement) {
+    this._scrollElement = element;
   }
 
   render() {
-    const { threads, threadOrder, changeThreadOrder, height } = this.props;
+    const { threads, threadOrder, changeThreadOrder, height, hiddenThreads } = this.props;
 
     return (
       <div className='timelineViewTimelines'>
         <div className='timelineViewTimelinesScroller'
-             ref={element => {
-               this.scrollElement = element;
-             }}>
+             ref={this._setScrollElementRef}>
```

Comment:
My 2 cents here is that it's quite useless to have a separate method for this, and this artificially increase the size of the file, making it less readable.

But this is not a blocker :)